contracts/tokenomics/generator/src/contract.rs

```diff
@@ -171,6 +188,10 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
     match msg {
+        ExecuteMsg::KickUserBoost { user, generators } => kick_user_boost(deps, user, generators),
```

Comment:
Imo, 'Kick a user' just means recalculate his virtual amount and if it becomes 0 - do not pay him reward from now. This could be done by usual business logic in the Checkpoint function. So, I think we do not need this separate function.